### PR TITLE
Backwards ops supporting broadcasting

### PR DESCRIPTION
**Pull request by marcelbischoff**
* Adding adjoint operation to broadcasting adBC()
* Backwards is working with broadcasting for add, sub, mul, div, pow
* Enabled Testing on backward for broadcasting CPU (note div cannot be registered)
 
~~Needs to be tested.~~ Passes all tests of #106

### CODE REVIEW

**Comment by adamritter** [comment]
- sum(axis=...).reshape(...) is done so many times, I think it would be nicer to put it in a function

**Comment by marcelbischoff** [comment]
> sum(axis=...).reshape(...) is done so many times, I think it would be nicer to put it in a function

Definitely! First, need to update tests, though.

**Comment by adamritter** [comment]
Cool, now it's so easy to understand that it's trivial to implement for GPU ops as well, thanks!

**Comment by marcelbischoff** [comment]
> Cool, now it's so easy to understand that it's trivial to implement for GPU ops as well, thanks!

Just that you want a good sum reduction on GPU for speed!?

**Comment by adamritter** [comment]
I guess right now so many kernels are far from optimal that benchmarking is
more important than a good sum reduction (it seems that George doesn't care
about optimal speed, that would be fun to do in a fork, like fasttinygrad
:) )

The conv kernels are also doing simple addition (especially addition to
memory, which is probably slower than using a register).

**Comment by marcelbischoff** [comment]
No, I think he first want feature completeness. It is so much easier to optimize something if you already have something that works!

**Comment by adamritter** [comment]
It's impossible to be fast and stay under 1000 lines. I saw some fast
matmul kernels and they take hundreds of lines....of course they can be
shortened as well, but there are always tradeoffs to take.

File: tinygrad/ops.py [hunk 1]
```diff
@@ -4,25 +4,31 @@
 from .tensor import Function, register
 
 # ************* basic ops *************
+def adBC(out, in_sh): #adjoint operation to broadcast is sum. Need to sum all axis with 1 = in_sh[i] < out.shape[i] 
+    return out.sum(axis=tuple([i for i in range(len(in_sh)) if in_sh[i]==1 and out.shape[i]>1])).reshape(in_sh)
```
Comment (geohot) [inline tinygrad/ops.py:8]: Spacing

**Review by geohot — COMMENTED** [review]
Very nice, fix spacing and I’ll merge!

File: tinygrad/ops.py (hunk 1 above)
Comment (marcelbischoff) [reply tinygrad/ops.py]: Fixed (also fixed my `.vimrc`)

**Pull request merged**